predicators/spot_utils/perception/object_detection.py

```diff
@@ -350,6 +349,36 @@ def _get_pose_from_segmented_bounding_box(
     return world_frame_pose
 
 
+def get_object_center_pixel_from_artifacts(
```

Comment:
LOVE this new refactor! This will finally prevent the issue where we see an object, but then ask the user for a grasp 2 seconds later because we happened to miss the detection right when we're grasping. This should also speed things up quite a bit!